File: CS/XtraDiagram.CustomShapeProperties/DiagramShapeEx.cs
using DevExpress.Diagram.Core;
using DevExpress.Utils.Serializing;
using DevExpress.XtraDiagram;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace XtraDiagram.CustomShapeProperties {
    public class DiagramShapeEx : DiagramShape {
        [XtraSerializableProperty]
        public int DatabaseObjectID { get; set; }
        static DiagramShapeEx() {
            DiagramControl.ItemTypeRegistrator.Register(typeof(DiagramShapeEx));
        }
    }
}

File: VB/XtraDiagram.CustomShapeProperties/DiagramShapeEx.vb
Imports DevExpress.Diagram.Core
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraDiagram
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace XtraDiagram.CustomShapeProperties
    Public Class DiagramShapeEx
        Inherits DiagramShape

        <Browsable(False), XtraSerializableProperty> _
        Public Property DatabaseObjectID() As Integer
    End Class
End Namespace
